cpp/server/ct-server.cc

```diff
@@ -224,34 +225,50 @@ static const bool sign_dummy =
     RegisterFlagValidator(&FLAGS_tree_signing_frequency_seconds,
                           &ValidateIsPositive);
 
+void CleanUpEntries(ConsistentStore<LoggedCertificate>* store,
+                    const MasterElection* election) {
+  CHECK_NOTNULL(store);
+  CHECK_NOTNULL(election);
+  const steady_clock::duration period(
+      (seconds(FLAGS_cleanup_frequency_seconds)));
+  steady_clock::time_point target_run_time(steady_clock::now());
+
+  while (true) {
+    if (election->IsMaster()) {
+      const ScopedLatency sequencer_delete_latency(
+          sequencer_delete_latency_ms.ScopedLatency());
+      util::Status status(store->CleanupOldEntries());
+      if (!status.ok()) {
+        LOG(WARNING) << "Problem cleaning up old entries: " << status;
+      }
+    }
+
+    const steady_clock::time_point now(steady_clock::now());
+    while (target_run_time <= now) {
+      target_run_time += period;
+    }
+
+    std::this_thread::sleep_for(target_run_time - now);
+  }
+}
+
 void SequenceEntries(TreeSigner<LoggedCertificate>* tree_signer,
-                     ConsistentStore<LoggedCertificate>* store,
                      const MasterElection* election) {
+  CHECK_NOTNULL(tree_signer);
+  CHECK_NOTNULL(election);
   const steady_clock::duration period(
       (seconds(FLAGS_sequencing_frequency_seconds)));
   steady_clock::time_point target_run_time(steady_clock::now());
 
   while (true) {
     if (election->IsMaster()) {
-      const ScopedLatency sequence_run_latency(
```

Comment:
Just because it'd now be measuring the same thing as `sequencer_delete_latency_ms` give or take a couple of µs